Dart/src/com/jetbrains/lang/dart/psi/impl/DartReferenceImpl.java

```diff
@@ -140,9 +140,12 @@ public DartClassResolveResult resolveDartClass() {
       }
     }
     if (this instanceof DartCascadeReferenceExpression) {
-      DartReference[] children = PsiTreeUtil.getChildrenOfType(this, DartReference.class);
-      if (children != null && children.length == 2) {
-        return children[0].resolveDartClass();
+      PsiElement parent = this.getParent();
+      if (parent instanceof DartValueExpression) {
+        DartReference child = (DartReference) parent.getFirstChild();
```

Comment:
Dangerous unchecked cast.
In general, getFirstChild() may be null (not here of course), whitespace, comment (unlikely here, but still), throwExpression (which is not a DartReference).
Where possible try to use more specific methods (DartValueExpression.getExpressionList()) in this case instead of general ones like PsiElement.getFirstChild().